burn-core/src/optim/simple/adaptor.rs

```diff
@@ -45,8 +60,22 @@ where
     type Record = HashMap<ParamId, AdaptorRecord<O, B::InnerBackend>>;
 
     fn step(&mut self, lr: LearningRate, module: M, mut grads: GradientsParams) -> M {
-        let mut mapper =
-            SimpleOptimizerMapper::<M, B, O>::new(&self.optim, &mut self.records, &mut grads, lr);
+        if let Some(ref g_clipper) = self.gradient_clipper {
+            grads.iter_mut().for_each(|(id, grad)| {
+                let grad_tensor: Tensor<B, D> =
+                    grad.downcast().expect("Failed to downcast gradient");
+                let clipped_grad = g_clipper.clip_gradient(grad_tensor);
+                grads.register(id.clone(), clipped_grad);
+            })
+        }
```

Comment:
@nathanielsimard Can I get your opinion on the best way to do this/if this is necessary for SimpleOptimizer? I think that if there's a gradient clipper being used with a simple optimizer, it's best and easiest to apply it to all the gradients. But I'm not sure what the best way to iterate over and subsequently clip all those gradients is. I've added iter_mut implementations to TensorContainer and GradientsParams because it doesn't look like there's a way to get all the gradients.

Reply:
I don't think it's a good idea to iterate over all tensor to apply the clipper, since tensor are of different types (different dimensions) and you have to use the parameter id linked to the module to determine witch tensor type it is.

I think you can just call the clipper in the `map` function of the `SimpleOptimizerMapper` struct, just before `optimizer.step` is called, or directly in the `step` function of each optimizer implementation.